common-http/src/main/java/io/cdap/common/http/HttpResponse.java

```diff
@@ -45,6 +59,25 @@
     this.headers = headers;
   }
 
+  HttpResponse(HttpURLConnection conn) throws IOException {
+    this.conn = conn;
+    this.responseCode = conn.getResponseCode();
+    this.responseMessage = conn.getResponseMessage();
+    this.headers = parseHeaders(conn.getHeaderFields());
+    this.inputStream = isSuccessful(responseCode) ? conn.getInputStream() : conn.getErrorStream();
+    this.responseBody = getResponseBodyFromStream();
+  }
+
+  HttpResponse(HttpURLConnection conn, HttpContentConsumer consumer) throws IOException {
```

Comment:
Make consumer `@Nullable` so that it can be called from another constructor with `null`

Reply:
done